CMake/RealmCore.cmake

```diff
@@ -237,9 +237,9 @@ function(build_realm_sync sync_directory)
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION_RELEASE ${sync_server_library_release})
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION ${sync_server_library_release})
     if(APPLE)
-        set_property(TARGET realm-sync-server PROPERTY INTERFACE_LINK_LIBRARIES ${FOUNDATION} ${SECURITY})
+        set_property(TARGET realm-sync-server PROPERTY INTERFACE_LINK_LIBRARIES ${FOUNDATION} ${SECURITY} -lyaml-cpp)
```

Comment:
If we're going to do this then realm-sync should really be changed to _require_ yaml-cpp, rather than only using it if it can find it. As it stands we're in this weird state where realm-sync will use yaml-cpp if it can find it, but there's no way for clients of realm-sync to know whether they need to link against yaml-cpp.